### PR TITLE
qdated 0.53 (new formula)

### PR DESCRIPTION
**Pull request by lukasbestle**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/qdated.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Qdated < Formula
+  desc "Create and verify timestamped email addresses for qmail"
+  homepage "https://web.archive.org/web/http://www.palomine.net/qdated/"
+  url "https://web.archive.org/web/http://www.palomine.net/qdated/qdated-0.53.tar.gz"
+  sha256 "002391c656c705838965cab8231261f2474b26711c03e24cf0ecf2978b3cf934"
+
+  bottle do
+    cellar :any_skip_relocation
+  end
```
Comment (chenrui333) [inline Formula/qdated.rb:9]: maybe uneeded?
```suggestion
  bottle :unneeded
```

Comment (chenrui333) [reply Formula/qdated.rb:9]: if we need bottle artifact, we should remove this block.

Comment (lukasbestle) [reply Formula/qdated.rb:9]: Thanks for your feedback!

The bottle is needed as it's a compiled formula. However the formula doesn't link against anything, so `:any_skip_relocation` is supported.

If the block should be removed completely: How should I add `:any_skip_relocation` then?

File: Formula/qdated.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class Qdated < Formula
+  desc "Create and verify timestamped email addresses for qmail"
+  homepage "https://web.archive.org/web/http://www.palomine.net/qdated/"
+  url "https://web.archive.org/web/http://www.palomine.net/qdated/qdated-0.53.tar.gz"
```
Comment (SMillerDev) [inline Formula/qdated.rb:4]: Is there no current webpage available?

File: Formula/qdated.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/qdated.rb:9]: `:any_skip_relocation` is added by the bot if it determines that that's an option

File: Formula/qdated.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    bin.install "qdated-#{version}/command/qdated-check"
+    bin.install "qdated-#{version}/command/qdated-makekey"
+    bin.install "qdated-#{version}/command/qdated-now"
+
+    File.rename "qdated-#{version}/package/LICENSE", "LICENSE"
+  end
+
+  test do
+    require "English"
```
Comment (SMillerDev) [inline Formula/qdated.rb:24]: Why is this required?

File: Formula/qdated.rb (hunk 1 above)
Comment (lukasbestle) [reply Formula/qdated.rb]: ✅

File: Formula/qdated.rb (hunk 2 above)
Comment (lukasbestle) [reply Formula/qdated.rb:4]: The author's site seems to be down and the code or docs haven't been published anywhere else.

File: Formula/qdated.rb (hunk 3 above)
Comment (lukasbestle) [reply Formula/qdated.rb:20]: The test uses the `$CHILD_STATUS` variable, which is provided by the `English` library. I was using `$?` before, but `brew audit` told me to use `$CHILD_STATUS` from `English` instead.

File: Formula/qdated.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/qdated.rb:4]: I think this violates the "must have a homepage" requirement for acceptable formulae here: https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

Any of the @Homebrew/core maintainers think otherwise?

Comment (lukasbestle) [reply Formula/qdated.rb:4]: Hm, but there are several formulae with homepages on the Web Archive:

```
tcping.rb:      homepage "https://web.archive.org/web/20171102193822/linuxco.de/tcping/tcping.html"
einstein.rb:    homepage "https://web.archive.org/web/20120621005109/games.flowix.com/en/index.html"
apachetop.rb:   homepage "https://web.archive.org/web/20170809160553/freecode.com/projects/apachetop"
compface.rb:    homepage "https://web.archive.org/web/20170720045032/freecode.com/projects/compface"
mpack.rb:       homepage "https://web.archive.org/web/20190220145801/ftp.andrew.cmu.edu/pub/mpack/"
trafshow.rb:    homepage "https://web.archive.org/web/20130707021442/soft.risp.ru/trafshow/index_en.shtml"
pakchois.rb:    homepage "https://web.archive.org/web/www.manyfish.co.uk/pakchois/"
f3.rb:          homepage "https://web.archive.org/web/20200110163924/oss.digirati.com.br/f3/"
libelf.rb:      homepage "https://web.archive.org/web/20181111033959/www.mr511.de/software/english.html"
shorten.rb:     homepage "https://web.archive.org/web/20180903155129/www.etree.org/shnutils/shorten/"
zdelta.rb:      homepage "https://web.archive.org/web/20150804051750/cis.poly.edu/zdelta/"
cronolog.rb:    homepage "https://web.archive.org/web/20140209202032/cronolog.org/"
dvorak7min.rb:  homepage "https://web.archive.org/web/dvorak7min.sourcearchive.com/"
redir.rb:       homepage "https://web.archive.org/web/20190817033513/sammy.net/~sammy/hacks/"
colorsvn.rb:    homepage "https://web.archive.org/web/20170725092001/colorsvn.tigris.org/"
delta.rb:       homepage "https://web.archive.org/web/20170805142100/delta.tigris.org/"
ibex.rb:        homepage "https://web.archive.org/web/20190826220512/www.ibex-lib.org/"
shntool.rb:     homepage "https://web.archive.org/web/20180903160820/etree.org/shnutils/shntool/"
ossp-uuid.rb:   homepage "https://web.archive.org/web/www.ossp.org/pkg/lib/uuid/"
libcapn.rb:     homepage "https://web.archive.org/web/20181220090839/libcapn.org/"
tal.rb:         homepage "https://web.archive.org/web/20160406172703/https://thomasjensen.com/software/tal/"
xml2.rb:        homepage "https://web.archive.org/web/20160730094113/www.ofb.net/~egnor/xml2/"
libghthash.rb:  homepage "https://web.archive.org/web/20170824230514/www.bth.se/people/ska/sim_home/libghthash.html"
pbzip2.rb:      homepage "https://web.archive.org/web/20180226093549/compression.ca/pbzip2/"
sendemail.rb:   homepage "https://web.archive.org/web/20191013154932/caspian.dotconf.net/menu/Software/SendEmail/"
alac.rb:        homepage "https://web.archive.org/web/20150319040222/craz.net/programs/itunes/alac.html"
jnettop.rb:     homepage "https://web.archive.org/web/20161127214942/jnettop.kubs.info/wiki/"
netris.rb:      homepage "https://web.archive.org/web/20071223041235/www.netris.be/"
lv.rb:          homepage "https://web.archive.org/web/20160310122517/www.ff.iij4u.or.jp/~nrt/lv/"
jerm.rb:        homepage "https://web.archive.org/web/20160719014241/bsddiary.net/jerm/"
tta.rb:         homepage "https://web.archive.org/web/20100131140204/true-audio.com/"
contacts.rb:    homepage "https://web.archive.org/web/20181108222900/gnufoo.org/contacts/contacts.html"
ecm.rb:         homepage "https://web.archive.org/web/20140227165748/www.neillcorlett.com/ecm/"
imap-uw.rb:     homepage "https://web.archive.org/web/20191028114408/https://www.washington.edu/imap/"
gf-complete.rb: homepage "https://web.archive.org/web/20191024182742/jerasure.org/"
arp-sk.rb:      homepage "https://web.archive.org/web/20180223202629/sid.rstack.org/arp-sk/"
rtptools.rb:    homepage "https://web.archive.org/web/20190924020700/www.cs.columbia.edu/irt/software/rtptools/"
procmail.rb:    homepage "https://web.archive.org/web/20151013184044/procmail.org/"
udptunnel.rb:   homepage "https://web.archive.org/web/20161224191851/www.cs.columbia.edu/~lennox/udptunnel/"
popt.rb:        homepage "https://web.archive.org/web/20190425081726/rpm5.org/"
pgtune.rb:      homepage "https://web.archive.org/web/20190717075914/pgfoundry.org/projects/pgtune"
aggregate.rb:   homepage "https://web.archive.org/web/20160716192438/freecode.com/projects/aggregate/"
jerasure.rb:    homepage "https://web.archive.org/web/20191024182742/jerasure.org/"
```

Comment (SMillerDev) [reply Formula/qdated.rb:4]: As far as I know all of those were admitted with a fully functioning homepage that went offline later. The fact that both the source and homepage are offline indicates to me that this is abandoned software that will never get (security) fixes.

Comment (lukasbestle) [reply Formula/qdated.rb:4]: That makes sense. Let's see what the others think, otherwise I'd be fine with creating a tap for this. :)

Comment (MikeMcQuaid) [reply Formula/qdated.rb:4]: Yeh, sorry, we have a higher standard for new formula than existing formula. I agree we should 👎 this for now 😭

Comment (vszakats) [reply Formula/qdated.rb:4]: I can second that these were updated (usually long) after the formula has been added and the official homepage became unavailable since. A newly added project without a working homepage is certainly not a good sign.

Either way, can we remove the `http://` prefix from the URL? It isn't necessary and can make the line appear when grepping for insecure links.

Comment (lukasbestle) [reply Formula/qdated.rb]: > Either way, can we remove the `http://` prefix from the URL? It isn't necessary and can make the line appear when grepping for insecure links.

@vszakats Sure, done.

If you think this can be merged, please do so, otherwise feel free to close this PR. :)

Comment (vszakats) [reply Formula/qdated.rb]: @lukasbestle Thank you for the fix! I'm not to judge the usefulness of `qdated`, it may still be useful tool without an active homepage.

Comment (lukasbestle) [reply Formula/qdated.rb]: It is! It's quite tiny and therefore it's very likely that it's just "done" (like `qmail`, the web server it is made for, which also hasn't been updated for years and doesn't need to).

However I understand that you need to have clear rules that tell you which formulae can be supported in core and which can't, and that's completely fine for me. After all this is a free open-source project and and you all do this in your free time.
